pkg/envoy/xdsutil.go

```diff
@@ -399,10 +399,9 @@ func GetPodFromCertificate(cn certificate.CommonName, kubecontroller k8s.Control
 
 // GetServiceIdentityFromProxyCertificate returns the ServiceIdentity information encoded in the XDS certificate CN
 func GetServiceIdentityFromProxyCertificate(cn certificate.CommonName) (identity.ServiceIdentity, error) {
-	var identity identity.ServiceIdentity
 	cnMeta, err := getCertificateCommonNameMeta(cn)
 	if err != nil {
-		return identity, err
+		return "", err
```

Comment:
would it be better to return an empty struct ?

Reply:
Funny thing is - I tried -- turns out our current iteration of `identity.ServiceIdentity` is a string type alias - so this works for now (`identity.ServiceIdentity{}` does not).

But one day I suspect `identity.ServiceIdentity` will be a struct and then we will be able to return an empty struct (or nil).